core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -534,6 +534,9 @@ private RexNode simplifyIsNull(RexNode a) {
     if (!a.getType().isNullable()) {
       return rexBuilder.makeLiteral(false);
     }
+    if (RexUtil.isNull(a)) {
```

Comment:
I see that RexUtil#isNull is used in various other places inside RexSimplify so it seems reasonable to use it here. However, there is also org.apache.calcite.plan.Strong#isNull(org.apache.calcite.rex.RexNode) which covers some other cases. I am wondering if it makes sense to somehow merge both (or use both)?

Reply:
There is also the following comment at line 444:
 // x IS NULL ==> FALSE (if x is not nullable)
I guess it should be updated to reflect the changes.

Reply:
It looks like `Strong#isNull` duplicates the efforts of `RexSimplify`, so I would suggest to keep `RexUtil#isNull`

Reply:
> What should the comment be updated to? The comment still applies to this patch, doesn't it?

I was thinking to add something like the following:
// x is NULL ==> TRUE (if x is null)

Reply:
> It looks like `Strong#isNull` duplicates the efforts of `RexSimplify`, so I would suggest to keep `RexUtil#isNull`

It's fine with me. 
Other than that, I checked the chages in the tests (sub-query.iq) and the simplifications seems correct. 
@vlsi it would be great if you could merge this since I need the fix (I did it also in my branch but I will revert) for another issue.

Reply:
Does this issue really hurt you?

I just thought this was a minor issue, and merging this would interfere with https://github.com/apache/calcite/pull/831